quickannotator/dsa_sdk.py

```diff
@@ -0,0 +1,151 @@
+import requests
+import urllib.parse
+import json
+
+class DSAClient:
+    # NOTE: may alternatively want to accept a token.
+    def __init__(self, base_url, api_key, duration=1):
+        """
+        Initialize the DSA SDK.
+
+        Args:
+            base_url (str): The base URL of the DSA server.
+            api_key (str, optional): The API key to generate the Girder authentication token.
+            duration (int, optional): The duration (in days) for which the token is valid.
+        """
+        self.base_url = base_url.rstrip('/')
+        self.token = None
+        try:
+            self.get_token_with_api_key(api_key, duration)
```

Comment:
should this set the token if successful?

Reply:
Currently the `get_token_with_api_key()` method internally sets `self.token` after a successful get. 

But what you're suggesting makes more sense - the `get_token_with_api_key` method should get the token and do nothing else. I've updated the code accordingly.